### DISCUSSION TITLE
Add `duration-0` and `delay-0` utilities

### DISCUSSION THREAD

**Discussion by MichaelAllenWarner** [tailwindlabs/tailwindcss#6031] [Ideas]
Having `duration-0` and `delay-0` utilities available would provide a native way to _conditionally_ zero-out the `transition-duration` and `transition-delay` properties (without turning the the transition off altogether). For example:

```html
<div class="transition-colors duration-200 delay-200 lg:delay-0">...</div>
```

**Comment by timohausmann** [comment]
Agree! `delay-0` is even used in the docs, but the class does not exist.
https://tailwindcss.com/docs/transition-delay#applying-conditionally

**Comment by jamiebuilds-signal** [comment]
> Moving from https://github.com/tailwindlabs/tailwindcss/discussions/10293

This is an occasional pattern I use:

```html
<div id="titlebar" class="... bg-transparent bg-indigo-600 transition-colors delay-300 active:delay-0" />
```

- Default state is `bg-transparent`
- Immediately changes the color to `bg-indigo-600` upon `active:` 
- Delays the transition back to default state of `bg-transparent` by `delay-300`

This is useful for making something feel instantly responsive, but less visually distracting when you're clicking it multiple times.

I found it surprising that Tailwind did not have this baked in

**Reply by MichaelAllenWarner** [reply]
Yes, that's been my usual use-case also: a non-zero value for the default state, and then a zero value for some other specified state (`active:` or `aria-expanded:` or whatever).

**Comment by jamiebuilds-signal** [comment]
Added here: https://github.com/tailwindlabs/tailwindcss/pull/10294
